_pytest/compat.py

```diff
@@ -82,7 +82,10 @@ def num_mock_patch_args(function):
     return len(patchings)
 
 
-def getfuncargnames(function, startindex=None):
+def getfuncargnames(function, startindex=None, cls=None):
+    if startindex is None and cls is not None:
```

Comment:
> it moves a new responsibility into getfuncargnames, and fills the code with a lot of conditionals that are strictly not needed at that place and interaction

Oh I see. I decided to move `cls` here so that callers didn't have to handle known that `cls` might affect the results of this call themselves; the previous code was splitting that responsibility between two places, `getfixtureinfo` in `_pytest/fixtures.py` and here (see the diff). IMHO if `getfuncargnames` also needs a `cls` argument to figure things out, it should receive it and not require callers to handle that special case themselves.

> while reviewing i noticed that this also creates a dependence on how things are named, it will work incorrectly in some cases

Which things? The function name or the argument names?